python/pyADflow.py

```diff
@@ -3298,8 +3298,11 @@ def solveAdjoint(self, aeroProblem, objective, forcePoints=None,
             # Now set the flags and possibly reset adjoint
             if self.adflow.killsignals.adjointfailed:
                 self.curAP.adjointFailed = True
-                # Reset stored adjoint
-                self.curAP.adflowData.adjoints[objective][:] = 0.0
+                # Reset stored adjoint if we want to skip
+                if self.getOption('skipafterfailedadjoint'):
+                    self.curAP.adflowData.adjoints[objective][:] = 0.0
+                else:
+                    self.curAP.adflowData.adjoints[objective] = psi
```

Comment:
Wouldn't this set the value of the adjoint using the failed solve? Isn't that a problem?

Reply:
Some recent ordering fix to the zipper mesh stuff can cause some existing adjoint solutions to stall. So this `'skipafterfailedadjoint'` option allow people to use the adjoint even it was not converged to the specified tolerance. The default is `True`, so it's only used when you specify it.